bgpd/bgp_route.c

```diff
@@ -414,6 +414,29 @@ bgp_info_cmp (struct bgp *bgp, struct bgp_info *new, struct bgp_info *exist,
   newattre = newattr->extra;
   existattre = existattr->extra;
 
+  if (exist->sub_type == BGP_ROUTE_NORMAL
+      && new->sub_type != BGP_ROUTE_NORMAL
+      && newattr->distance != 0) {
+    struct prefix *exist_prefix = &(exist->net->p);
+    u_char bgp_distance = bgp_distance_apply(exist_prefix, exist,
+          exist_prefix->family, SAFI_UNICAST, bgp);
```

Comment:
Let's follow existing style, exist_prefix.... should start under (exist_prefix on the line above